File: 1935-maximum-number-of-words-you-can-type/1935-maximum-number-of-words-you-can-type.cpp
class Solution {
public:
    int canBeTypedWords(string text, string brokenLetters) {
        // 用 unordered_set 存壞掉的字母，查詢 O(1)
        unordered_set<char> broken(brokenLetters.begin(), brokenLetters.end());

        stringstream ss(text); // 用 stringstream 切單字
        string word;
        int count = 0;

        while (ss >> word){
            // 逐字讀取
            bool canType = true;
            for (char ch : word){
                if (broken.count(ch)){
                    // 單字含有壞掉字母
                    canType = false;
                    break; // 提早跳出，省時間
                }
            }
            if (canType) count++;
        }
        return count;
    }
};

File: 1935-maximum-number-of-words-you-can-type/1935-maximum-number-of-words-you-can-type.py
class Solution:
    def canBeTypedWords(self, text: str, brokenLetters: str) -> int:
        # 把 text 切成單字
        words = text.split()
        # 把壞掉的字母存成 set，方便快速查詢
        broken = set(brokenLetters)

        count = 0 # 記錄可以打出的單字數量

        # 檢查每個單字
        for word in words:
            # 如果單字內「沒有」任何壞掉字母 → 可輸入
            if not any(ch in broken for ch in word):
                count += 1
        
        return count